### PR TITLE
update async commit related configurations

### PR DESCRIPTION
**Pull request by sticnarf**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Async commit and 1PC configurations have been moved from config file to system variables.

And there are known correctness issues about 1PC, so 1PC content is hidden.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@sticnarf Please involve a technical review~

**Comment by sticnarf** [comment]
PTAL @youjiali1995

**Review by youjiali1995 — COMMENTED** [review]
lgtm

File: system-variables.md [hunk 1]
```diff
@@ -1005,3 +1005,39 @@ explain select * from t where age=5;
...
+> **警告：**
+>
+> 当前该功能为实验特性，不建议在生产环境中使用。目前存在已知问题有：
+>
+> + 暂时与 [TiCDC](/ticdc/ticdc-overview.md) 不兼容，可能导致 TiCDC 运行不正常。
+> + 暂时与 [Compaction Filter](/tikv-configuration-file.md#enable-compaction-filter) 不兼容，共同使用时有小概率发生写丢失。
+
+- 作用域：SESSION | GLOBAL
+- 默认值：OFF
+- 该变量控制是否启用 Async Commit 特性，使事务两阶段提交的第二阶段于后台异步进行。开启本特性能降低事务提交的延迟。本特性与 TiDB Binlog 不兼容，开启 binlog 时本配置将没有效果。
```
Comment (TomShawn) [inline system-variables.md:1020]:
```suggestion
- 该变量控制是否启用 Async Commit 特性，使事务两阶段提交的第二阶段于后台异步进行。开启本特性能降低事务提交的延迟。本特性与 TiDB Binlog 不兼容，开启 TiDB Binlog 时本配置将不生效。
```

File: system-variables.md [hunk 2]
```diff
@@ -1005,3 +1005,39 @@ explain select * from t where age=5;
...
+- **注意：开启本特性时，默认不保证事务的外部一致性。具体请参考 [`tidb_guarantee_external_consistency`](#tidb_guarantee_external_consistency-从-v500-rc-版本开始引入) 系统变量。**
```
Comment (TomShawn) [inline system-variables.md:1021]:
```suggestion

> **注意：**
>
> 开启本特性时，默认不保证事务的外部一致性。具体请参考 [`tidb_guarantee_external_consistency`](#tidb_guarantee_external_consistency-从-v500-rc-版本开始引入) 系统变量。
```

File: system-variables.md [hunk 3]
```diff
@@ -1005,3 +1005,42 @@ explain select * from t where age=5;
...
+> 当前该功能为实验特性，不建议在生产环境中使用。目前存在已知问题有：
+>
+> + 暂时与 [TiCDC](/ticdc/ticdc-overview.md) 不兼容，可能导致 TiCDC 运行不正常。
+> + 暂时与 [Compaction Filter](/tikv-configuration-file.md#enable-compaction-filter) 不兼容，共同使用时有小概率发生写丢失。
+
+- 作用域：SESSION | GLOBAL
+- 默认值：OFF
+- 指定是否在只涉及一个 Region 的事务上使用一阶段提交特性。比起传统两阶段提交，一阶段提交能大幅降低事务提交延迟并提升吞吐。
+- **注意：开启本特性时，默认不保证事务的外部一致性。具体请参考 [`tidb_guarantee_external_consistency`](#tidb_guarantee_external_consistency-从-v500-rc-版本开始引入) 系统变量。**
+-->
```
Comment (TomShawn) [inline system-variables.md:1040]:
```suggestion
```

File: system-variables.md [hunk 4]
```diff
@@ -1005,3 +1005,27 @@ explain select * from t where age=5;
...
+> + 暂时与 [Compaction Filter](/tikv-configuration-file.md#enable-compaction-filter) 不兼容，共同使用时有小概率发生写丢失。
+
+- 作用域：SESSION | GLOBAL
+- 默认值：OFF
+- 该变量控制是否启用 Async Commit 特性，使事务两阶段提交的第二阶段于后台异步进行。开启本特性能降低事务提交的延迟。本特性与 TiDB Binlog 不兼容，开启 TiDB Binlog 时本配置将不生效。
+
+> **注意：**
+>
+> 开启本特性时，默认不保证事务的外部一致性。具体请参考 [`tidb_guarantee_external_consistency`](#tidb_guarantee_external_consistency-从-v500-rc-版本开始引入) 系统变量。
+
```
Comment (TomShawn) [inline system-variables.md:1025]:
```suggestion
```

**Review by TomShawn — APPROVED** [review]
LGTM

File: system-variables.md [hunk 5]
```diff
@@ -1005,3 +1005,26 @@ explain select * from t where age=5;
...
 - 这个变量表示是否追踪聚合函数的内存使用情况。当开启该功能时，聚合函数的内存使用情况会被统计，进而可能会造成整个 SQL 内存统计值超阈值 [`mem-quota-query`](/tidb-configuration-file.md#mem-quota-query)，然后被 [`oom-action`](/tidb-configuration-file.md#oom-action) 定义的行为影响。
+
+### `tidb_enable_async_commit` <span class="version-mark">从 v5.0.0-rc 版本开始引入</span>
+
+> **警告：**
+>
+> 当前该功能为实验特性，不建议在生产环境中使用。目前存在已知问题有：
+>
+> + 暂时与 [TiCDC](/ticdc/ticdc-overview.md) 不兼容，可能导致 TiCDC 运行不正常。
+> + 暂时与 [Compaction Filter](/tikv-configuration-file.md#enable-compaction-filter) 不兼容，共同使用时有小概率发生写丢失。
```
Comment (scsldb) [inline system-variables.md:1016]: 当前该功能为实验特性，不建议在生产环境中使用。目前存在已知问题有
修改成：
实验特性，不推荐在生产环境中使用。目前已知不兼容性，如下：

本特性与 TiDB Binlog 不兼容，开启 TiDB Binlog 时本配置将不生效
修改成：开启 TiDB Binlog 时，该变量值不生效。

**Pull request merged**